### PR TITLE
formulae: specify bin path in test block

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
followup of 
- #122653
- #122667

### CODE REVIEW

**Comment by chenrui333** [comment]
still need a bit of work.

File: Formula/at-spi2-atk.rb [hunk 1]
```diff
@@ -84,8 +84,9 @@ def install
       }
     EOS
 
-    pkg_config_cflags = shell_output("pkg-config --cflags --libs atk-bridge-2.0 glib-2.0 atk").chomp.split
-    system ENV.cc, "test.c", *pkg_config_cflags, "-o", "test"
+    pkg_config_cmd = Formula["pkg-config"].opt_bin/"pkg-config --cflags --libs atk-bridge-2.0 glib-2.0 atk"
```
Comment (carlocab) [inline Formula/at-spi2-atk.rb:87]: We don't need to specify the full path to `pkg-config`. We can just use the one in `PATH`. We only need to specify the path for executables provided by the formula.

Comment (chenrui333) [reply Formula/at-spi2-atk.rb:87]: we have mixed usage of pkg-config in the repo (some do specify the pkg-path, some do not)

Comment (SMillerDev) [reply Formula/at-spi2-atk.rb:87]: But it doesn't matter except in the pkg-config formula. We only need to ensure that we're not testing the system version of our packages.

**Pull request closed without merging**